### PR TITLE
refactor: Extract message "type"s out into constants to help prevent string mismatches

### PR DESCRIPTION
**Pull request by Josmithr**
Adds message "type" constants, and updates usages of string literals to use these constants instead. Should help prevent bugs like the one fixed in #14974 going forward.

Also renamed "AudienceClientMetaData" to "AudienceClientMetadata" to be consistent with other usages of the term "metadata".

### CODE REVIEW

File: packages/tools/client-debugger/client-debugger-view/src/components/ContainerHistoryView.tsx [hunk 1]
```diff
@@ -40,7 +42,7 @@ export function ContainerHistoryView(props: ContainerHistoryProps): React.ReactE
 		 * Handlers for inbound messages related to the registry.
 		 */
 		const inboundMessageHandlers: InboundHandlers = {
-			["CONTAINER_STATE_HISTORY"]: (untypedMessage) => {
+			[ContainerStateChangeMessageType]: (untypedMessage) => {
```
Comment (alexvy86) [inline packages/tools/client-debugger/client-debugger-view/src/components/ContainerHistoryView.tsx:45]:
```suggestion
			[ContainerStateHistoryMessageType]: (untypedMessage) => {
```
I think?

**Review by alexvy86 — APPROVED** [review]
❤️ 

Nit: I start to get the feeling that the amount of boilerplate for the messaging piece is spiraling out of control. I think I would've centralized the message types instead of using individual constants (maybe a `MessageTypes` class with nothing but `public static readonly` fields). Thoughts?

File: packages/tools/client-debugger/client-debugger-view/src/components/ContainerHistoryView.tsx (hunk 1 above)
Comment (Josmithr) [reply packages/tools/client-debugger/client-debugger-view/src/components/ContainerHistoryView.tsx]: Oof, yup. Good catch! Intellisense fail.

**Comment by Josmithr** [comment]
> ❤️
> 
> Nit: I start to get the feeling that the amount of boilerplate for the messaging piece is spiraling out of control. I think I would've centralized the message types instead of using individual constants (maybe a `MessageTypes` class with nothing but `public static readonly` fields). Thoughts?

Yeah, after doing the refactoring exercise, I agree. I think I still like the idea of grouping the messages by category (audience, telemetry, container, etc.), but maybe a namespace / class for each category would be a reasonable compromise?

**Comment by alexvy86** [comment]
> Yeah, after doing the refactoring exercise, I agree. I think I still like the idea of grouping the messages by category (audience, telemetry, container, etc.), but maybe a namespace / class for each category would be a reasonable compromise?

I could see `MessageTypes.Telemetry.<*>`, `MessageTypes.Audience.<*>`, etc. Instinctively would prefer that over separate classes for each area. But thinking more thoroughly about back/forward-compat implications of the choice seems worth it.

**Comment by Josmithr** [comment]
> > Yeah, after doing the refactoring exercise, I agree. I think I still like the idea of grouping the messages by category (audience, telemetry, container, etc.), but maybe a namespace / class for each category would be a reasonable compromise?
> 
> I could see `MessageTypes.Telemetry.<*>`, `MessageTypes.Audience.<*>`, etc. Instinctively would prefer that over separate classes for each area. But thinking more thoroughly about back/forward-compat implications of the choice seems worth it.

Yeah, the compatibility side of things definitely warrants further consideration. I think I'll merge this as is for now (since it at least gives us a bit more type safety, even if it generates more boilerplate), and tackle cleaning up boilerplate separately.

**Pull request merged**